src/NzbDrone.Core/Notifications/Trakt/Trakt.cs

```diff
@@ -137,5 +120,105 @@ public void RefreshToken()
                 _logger.Warn($"Error refreshing trakt access token");
             }
         }
+
+        private string MapMediaType(Source source)
+        {
+            var traktSource = string.Empty;
+
+            switch (source)
+            {
+                case Source.BLURAY:
+                    traktSource = "bluray";
+                    break;
+                case Source.WEBDL:
+                    traktSource = "digital";
+                    break;
+                case Source.WEBRIP:
+                    traktSource = "digital";
+                    break;
+                case Source.DVD:
+                    traktSource = "dvd";
+                    break;
+                case Source.TV:
+                    traktSource = "dvd";
+                    break;
+            }
+
+            return traktSource;
+        }
+
+        private string MapResolution(int resolution)
+        {
+            var traktResolution = string.Empty;
+
+            switch (resolution)
+            {
+                case 2160:
+                    traktResolution = "uhd_4k";
+                    break;
+                case 1080:
+                    traktResolution = "hd_1080p";
+                    break;
+                case 720:
+                    traktResolution = "hd_720p";
+                    break;
+                case 576:
+                    traktResolution = "sd_576p";
+                    break;
+                case 480:
+                    traktResolution = "sd_480p";
+                    break;
+            }
+
```

Comment:
Not sure how it would look in C#, but this is the referenced code to determine if `hd_1080`, `sd_576`, and `sd_480` are progressive or interlaced (p or i). Important to note that Trakt does not have an interlaced option for `uhd_4k` or `hd_720p`

```
    radarr_resolution = movie['movieFile']['quality']['quality']['resolution']
    scan_type = movie['movieFile']['mediaInfo']['scanType']
    resolution_mapping = {
        2160:"uhd_4k",
        1080:"hd_1080",
        720:"hd_720p",
        480:"sd_480",
        576:"sd_576"
        }
    resolution = resolution_mapping.get(radarr_resolution, None)
    if resolution in ["hd_1080", "sd_480", "sd_576"]:
        if scan_type in ['Interlaced', 'MBAFF', 'PAFF']:
            resolution = '{}i'.format(resolution)
        else:
            resolution = '{}p'.format(resolution)
```